### ISSUE TITLE
localhost and non-localhost in nodelist causes hang

### ISSUE THREAD

**Issue by mprobson** [UIUC-PPL/charm#547]
*Original issue: https://charm.cs.illinois.edu/redmine/issues/547*

---
If a user specifies (at least) two hosts, one of which is localhost and the other is not then Charm will silently hang. From my understanding, this is due to the name resolution and loopback rules. Basically the host table is filled with a loopback even on the remote nodes instead of pointing to the node where Charm was launched. Phil's suggested solution is to check if you have both a loopback and a non-loopback address in the node table and abort/throw an error.

Example nodelist file:
group main
host localhost
host truth

I've tested this in both SMP and non-SMP modes and it fails (it shouldn't really matter, just being thorough).

**Comment by mprobson** [comment]
*Original date: 2014-10-22 02:58:45*

---
Once merged, chamrun should now detect when "host localhost" is used with "host <anything-but-localhost>" and exit printing an error. I tested this using several nodelists which included various permutations and counts of localhosts and non-local hosts:
* local, non-local
* local, local, non-local, non-local
* non-local, non-local, local local
* non-local, local, non-local

And they all failed in the correct place. Additionally, I tried this for nodelists with multiple groups and used the non-default group (main) as well.

I do have a question though: if you're on faith and your node list is host localhost then host faith this will run. Should we allow this case? It will complicate the code but isn't strictly wrong?
